python/dgl/graph.py

```diff
@@ -1710,6 +1817,41 @@ def edata(self):
         """
         return self.edges[:].data
 
+    @property
+    def batch_size(self):
+        """Number of graphs in this batch.
+
+        Returns
+        -------
+        int
+            Number of graphs in this batch."""
+        return self._batch_size
+
+    @property
+    def batch_num_nodes(self):
+        """Number of nodes of each graph in this batch.
+
+        Returns
+        -------
+        list
+            Number of nodes of each graph in this batch."""
+        if self._batch_num_nodes is None:
+            return [self.number_of_nodes()]
+        else:
+            return self._batch_num_nodes
```

Comment:
Put `if ... is None` to constructor.

Reply:
If so, `batch_num_nodes` would return wrong results after we mutate on the graph.